### ISSUE TITLE
can parser types: values with more than 1 byte are generated wrong 

### ISSUE THREAD

**Issue by AlejoPm03** [ZeniteSolar/CAN_IDS#10]
when values are separated in bytes all parts are with the total size, instead of the size of the part
Example:
```C
// Auxiliar Battery Voltage
typedef struct
{
    union {
        uint8_t raw[8];
        struct {
            uint8_t signature;  // Senders signature. Units: 
            uint16_t batvoltage_l;  // Battery Voltage, byte low. Units: V/100
            uint16_t batvoltage_h;  // Battery Voltage, byte high. Units: V/100
            uint16_t position_l;  // Tail Poisition, byte low. Units: °/100
            uint16_t position_h;  // Tail Poisition, byte high. Units: °/100
            uint16_t batcurrent_l;  // Battery Current, byte low. Units: A/100
            uint16_t batcurrent_h;  // Battery Current, byte high. Units: A/100
        };
    };
} can_mde22_steeringbat_measurements_msg_t;
```

This should be:
```C
// Auxiliar Battery Voltage
typedef struct
{
    union {
        uint8_t raw[8];
        struct {
            uint8_t signature;  // Senders signature. Units: 
            uint16_t batvoltage;  // Battery Voltage. Units: V/100
            uint16_t position;  // Tail Poisition. Units: °/100
            uint16_t batcurrent;  // Battery Current. Units: A/100
        };
    };
} can_mde22_steeringbat_measurements_msg_t;
```

Or:
```C
// Auxiliar Battery Voltage
typedef struct
{
    union {
        uint8_t raw[8];
        struct {
            uint8_t signature;  // Senders signature. Units: 
            union{
                uint16_t batvoltage;
                struct{
                    uint8_t batvoltage_l;  // Battery Voltage, byte low. Units: V/100
                    uint8_t batvoltage_h;  // Battery Voltage, byte high. Units: V/100
                };
            };
        };
    };
} can_mde22_steeringbat_measurements_msg_t;
```

**Comment by joaoantoniocardoso** [comment]
Nice catch, I think the latter alternative is best.

**Comment by AlejoPm03** [comment]
Closed by: 91b9c4cbdf0e0421772fb67fbddd0a6dec0e9514
